lusidtools/cocoon/utilities.py

```diff
@@ -260,27 +260,40 @@ def set_attributes_recursive(
     # For each of the attributes to populate
     for key in list(populate_attributes):
 
+        # Get the attribute type
+        attribute_type = obj_attr[key]
+
         # If it is an additional attribute, populate it with the provided values and move to the next attribute
         if key in list(additional_attributes.keys()):
-            obj_init_values[key] = additional_attributes[key]
-            continue
+            # Handle identifiers provided within instrument definition (e.g. 'Bond', 'Future', etc.)
+            if (key, attribute_type) == ("identifiers", "dict(str, str)"):
+                obj_init_values[key] = {
+                    str_key: row[str_value["value"]]
+                    for str_key, str_value in mapping[key].items()
+                }
+                continue
+            else:
+                obj_init_values[key] = additional_attributes[key]
+                continue
```

Comment:
You will continue here no matter what, so you can put continue just once outside the if/else on line 277 and remove it on line 274.